src/vs/workbench/contrib/debug/browser/debugEditorActions.ts

```diff
@@ -341,13 +344,35 @@ export class SelectionToWatchExpressionsAction extends EditorAction {
 	async run(accessor: ServicesAccessor, editor: ICodeEditor): Promise<void> {
 		const debugService = accessor.get(IDebugService);
 		const viewsService = accessor.get(IViewsService);
+		const languageFeaturesService = accessor.get(ILanguageFeaturesService);
 		if (!editor.hasModel()) {
 			return;
 		}
 
-		const text = editor.getModel().getValueInRange(editor.getSelection());
+		let expression: string | undefined = undefined;
+
+		const model = editor.getModel();
+		const selection = editor.getSelection();
+
+		if (!selection.isEmpty()) {
+			expression = model.getValueInRange(selection);
+		} else {
+			const position = editor.getPosition();
+			const cancellation = new CancellationTokenSource();
+			const evaluatableExpression = await getEvaluatableExpressionAtPosition(languageFeaturesService, model, position, cancellation.token);
```

Comment:
You can actually use `CancellationToken.None` or make the cancellation token optional

Reply:
Done. Marked the parameter as optional (with default to `CancellationToken.None`) for more convenient usages in future.